packages/evo-blockmodel/src/evo/blockmodel/client.py

```diff
@@ -0,0 +1,437 @@
+from __future__ import annotations
+
+import asyncio
+import uuid
+from uuid import UUID
+
+from evo import logging
+from evo.common import APIConnector, BaseAPIClient, Environment, HealthCheckType, ICache, ServiceHealth
+from evo.common.data import ServiceUser
+from evo.common.utils import get_service_health
+
+from ._types import Table
+from ._utils import convert_dtype, extract_payload
+from .data import BaseGridDefinition, BlockModel, RegularGridDefinition, Version
+from .endpoints import models
+from .endpoints.api import ColumnOperationsApi, JobsApi, OperationsApi, VersionsApi
+from .endpoints.models import (
+    AnyUrl,
+    BBox,
+    BBoxXYZ,
+    BlockSize,
+    ColumnHeaderType,
+    CreateData,
+    GeometryColumns,
+    JobErrorPayload,
+    JobResponse,
+    JobStatus,
+    Location,
+    OutputOptionsParquet,
+    QueryCriteria,
+    QueryDownload,
+    Rotation,
+    RotationAxis,
+    Size3D,
+    SizeOptionsRegular,
+)
+from .exceptions import CacheNotConfiguredException, JobFailedException, MissingColumnInTable
+from .io import BlockModelDownload, BlockModelUpload, get_cache_location_for_upload
+
+logger = logging.getLogger("blockmodel.client")
+
+__all__ = [
+    "BlockModelServiceClient",
+]
+
+
+def _job_id_from_url(job_url: AnyUrl) -> UUID:
+    job_id = job_url.path.split("/")[-1]
+    return UUID(job_id)
+
+
+def _version_from_model(version: models.Version) -> Version:
+    return Version(
+        bm_uuid=version.bm_uuid,
+        version_id=version.version_id,
+        version_uuid=version.version_uuid,
+        created_at=version.created_at,
+        created_by=ServiceUser.from_model(version.created_by),
+        comment=version.comment,
+        bbox=version.bbox,
+        base_version_id=version.base_version_id,
+        parent_version_id=version.parent_version_id,
+        columns=version.mapping.columns,
+        geoscience_version_id=version.geoscience_version_id,
+    )
+
+
+_GEOMETRY_COLUMNS = {"i", "j", "k", "x", "y", "z"}
+
+
+class BlockModelServiceClient(BaseAPIClient):
```

Comment:
For consistency with the other clients, I believe this should be:

```suggestion
class BlockModelAPIClient(BaseAPIClient):
```

Reply:
That makes sense. I'll push through an update once I hear back about the decision about the naming.